Makefile

```diff
@@ -340,7 +340,7 @@ build: generate fmt vet ## Build manager binary.
 
 run: export LOG_LEVEL = debug
 run: export LOG_MODE = development
-run: export OPERATOR_NAMESPACE = kuadrant-system
+run: export OPERATOR_NAMESPACE = $(shell kubectl config view --minify -o jsonpath='{..namespace}')
```

Comment:
This gives an empty value for me for a kind and crc cluster, so it will panic for most people using `make run` here unless it's just me 🤔 
https://github.com/Kuadrant/kuadrant-operator/blob/128ea20995e1b81fb4f26c9b63e8c175ee68e17e/controllers/state_of_the_world.go#L158

```json
{
    "kind": "Config",
    "apiVersion": "v1",
    "preferences": {},
    "clusters": [
        {
            "name": "api-crc-testing:6443",
            "cluster": {
                "server": "https://api.crc.testing:6443",
                "certificate-authority-data": "DATA+OMITTED"
            }
        }
    ],
    "users": [
        {
            "name": "developer/api-crc-testing:6443",
            "user": {
                "token": "REDACTED"
            }
        }
    ],
    "contexts": [
        {
            "name": "/api-crc-testing:6443/developer",
            "context": {
                "cluster": "api-crc-testing:6443",
                "user": "developer/api-crc-testing:6443"
            }
        }
    ],
    "current-context": "/api-crc-testing:6443/developer"
}
```

Reply:
The command line `kubectl config view --minify -o jsonpath='{..namespace}'` is pretty common way to parse the current context's namespace. I do not think there is anything wrong with it. 

Your kubeconfig does not have namespace set on the current context. This is a portion of my kubeconfig

```yaml
contexts:
- context:                                                                 
    cluster: mycluster:443                   
    namespace: kuadrant-system                                             
    user: eguzki/mycluster:443               
  name: kuadrant-system/mycluster:443/eguzki 
current-context: kuadrant-system/mycluster:443/eguzki 
```

Maybe `oc project kuadrant-system` will help??

I will try to enhance makefile command to handle error when the namespace is not available.

Reply:
added

Reply:
Hmm, yeah adding `oc project kuadrant-system` will add the `namespace` to the project, but without it there no namespace in the context

Reply:
Yeah, I don't know about this change tbh. I'm getting the following whenever I run `make run` on a fresh **kind** cluster made using `make local-env-setup`, so this will affect the `make run` command once merged to main.
```
🚨 namespace could not be parsed from kubeconfig 💥
```
This does get set on a **CRC** cluster as I was interfacting with the cluster by following the verification steps. So I assume this gets set typically when interacting with an openshift type cluster unless I'm missing something 🤔 

TBH the env var has a default now also so we probably don't need this set in the makefile also unless people want to explicitly change it
https://github.com/Kuadrant/kuadrant-operator/blob/c194aa4144349fc4f5be61cb549fa2042ca3e49c/controllers/state_of_the_world.go#L35

Reply:
Sorry, I wanted to read the namespace from the kubeconfig context, but turns out that in local env (kind), there is no namespace for the current context. 

Reverting to previous state with the addition of adding `OPERATOR_NAMESPACE` makefile variable to override default value (`kuadrant-system`).

Reply:
Actually, I think that for `make run`, the `OPERATOR_NAMESPACE` should be:
* When `OPERATOR_NAMESPACE` makefile variable is set, it takes that value. Else
* namespace from kubeconfig current context. If empty, then
* default from `KUADRANT_NAMESPACE` makefile variable, i.e. `kuadrant-system`.

But we can leave this for other PR.

Reply:
cc @Boomatang ^^